<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>rankListPage</class>
 <widget class="QWidget" name="rankListPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1200</width>
    <height>700</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Game Rankings</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
    stop:0 #0a0a2a, stop:0.5 #1a1a3a, stop:1 #0a0a2a);
    color: white;
    font-family: 'Smiley Sans';
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>20</number>
   </property>
   <property name="leftMargin">
    <number>30</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>30</number>
   </property>
   <property name="bottomMargin">
    <number>20</number>
   </property>
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="styleSheet">
      <string notr="true">QLabel {
    font: bold 40px 'Smiley Sans';
    color: #FFD700;
    padding: 15px;
    background: rgba(0, 0, 0, 150);
    border-radius: 15px;
    border: 3px solid #FFD700;
}</string>
     </property>
     <property name="text">
      <string>üèÜ GAME LEADERBOARD üèÜ</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="difficultyGroupBox">
     <property name="styleSheet">
      <string notr="true">QGroupBox {
    font: bold 22px 'Smiley Sans';
    color: #4FC3F7;
    border: 2px solid #4FC3F7;
    border-radius: 10px;
    margin-top: 20px;
    padding-top: 15px;
    background: rgba(0, 0, 0, 80);
}
QGroupBox::title {
    subcontrol-origin: margin;
    left: 20px;
    padding: 0 15px 0 15px;
}</string>
     </property>
     <property name="title">
      <string>SELECT DIFFICULTY</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>40</number>
      </property>
      <property name="leftMargin">
       <number>30</number>
      </property>
      <property name="topMargin">
       <number>35</number>
      </property>
      <property name="rightMargin">
       <number>30</number>
      </property>
      <property name="bottomMargin">
       <number>20</number>
      </property>
      <item>
       <widget class="QRadioButton" name="allRadio">
        <property name="styleSheet">
         <string notr="true">QRadioButton {
    color: white;
    font-size: 18px;
    padding: 10px 20px;
    border-radius: 8px;
    background: rgba(0, 0, 0, 60);
}
QRadioButton:hover {
    background: rgba(255, 215, 0, 0.2);
}
QRadioButton::indicator {
    width: 20px;
    height: 20px;
}
QRadioButton::indicator::unchecked {
    border: 2px solid #FFD700;
    border-radius: 10px;
}
QRadioButton::indicator::checked {
    border: 2px solid #FFD700;
    border-radius: 10px;
    background-color: #FFD700;
}</string>
        </property>
        <property name="text">
         <string>All Difficulties</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="easyRadio">
        <property name="styleSheet">
         <string notr="true">QRadioButton {
    color: white;
    font-size: 18px;
    padding: 10px 20px;
    border-radius: 8px;
    background: rgba(0, 0, 0, 60);
}
QRadioButton:hover {
    background: rgba(76, 175, 80, 0.2);
}
QRadioButton::indicator {
    width: 20px;
    height: 20px;
}
QRadioButton::indicator::unchecked {
    border: 2px solid #4CAF50;
    border-radius: 10px;
}
QRadioButton::indicator::checked {
    border: 2px solid #4CAF50;
    border-radius: 10px;
    background-color: #4CAF50;
}</string>
        </property>
        <property name="text">
         <string>Easy</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="mediumRadio">
        <property name="styleSheet">
         <string notr="true">QRadioButton {
    color: white;
    font-size: 18px;
    padding: 10px 20px;
    border-radius: 8px;
    background: rgba(0, 0, 0, 60);
}
QRadioButton:hover {
    background: rgba(33, 150, 243, 0.2);
}
QRadioButton::indicator {
    width: 20px;
    height: 20px;
}
QRadioButton::indicator::unchecked {
    border: 2px solid #2196F3;
    border-radius: 10px;
}
QRadioButton::indicator::checked {
    border: 2px solid #2196F3;
    border-radius: 10px;
    background-color: #2196F3;
}</string>
        </property>
        <property name="text">
         <string>Medium</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="hardRadio">
        <property name="styleSheet">
         <string notr="true">QRadioButton {
    color: white;
    font-size: 18px;
    padding: 10px 20px;
    border-radius: 8px;
    background: rgba(0, 0, 0, 60);
}
QRadioButton:hover {
    background: rgba(244, 67, 54, 0.2);
}
QRadioButton::indicator {
    width: 20px;
    height: 20px;
}
QRadioButton::indicator::unchecked {
    border: 2px solid #F44336;
    border-radius: 10px;
}
QRadioButton::indicator::checked {
    border: 2px solid #F44336;
    border-radius: 10px;
    background-color: #F44336;
}</string>
        </property>
        <property name="text">
         <string>Hard</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="rankLayout">
     <property name="spacing">
      <number>40</number>
     </property>
     <item>
      <widget class="QGroupBox" name="totalRankGroup">
       <property name="styleSheet">
        <string notr="true">QGroupBox {
    font: bold 26px 'Smiley Sans';
    color: #FF9800;
    border: 2px solid #FF9800;
    border-radius: 10px;
    margin-top: 20px;
    padding-top: 15px;
    background: rgba(0, 0, 0, 100);
}
QGroupBox::title {
    subcontrol-origin: margin;
    left: 20px;
    padding: 0 15px 0 15px;
}</string>
       </property>
       <property name="title">
        <string>TOTAL RANKING</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QTextBrowser" name="ranklist">
          <property name="styleSheet">
           <string notr="true">QTextBrowser {
    background: rgba(0, 0, 0, 180);
    border: 2px solid #555;
    border-radius: 8px;
    padding: 15px;
    color: #E0E0E0;
    font-size: 16px;
    font-family: 'Consolas', monospace;
}
QScrollBar:vertical {
    background: rgba(0, 0, 0, 100);
    width: 12px;
    border-radius: 6px;
}
QScrollBar::handle:vertical {
    background: #FFD700;
    border-radius: 6px;
    min-height: 20px;
}</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="userRankGroup">
       <property name="styleSheet">
        <string notr="true">QGroupBox {
    font: bold 26px 'Smiley Sans';
    color: #00BCD4;
    border: 2px solid #00BCD4;
    border-radius: 10px;
    margin-top: 20px;
    padding-top: 15px;
    background: rgba(0, 0, 0, 100);
}
QGroupBox::title {
    subcontrol-origin: margin;
    left: 20px;
    padding: 0 15px 0 15px;
}</string>
       </property>
       <property name="title">
        <string>PERSONAL RECORDS</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QTextBrowser" name="userRanklist">
          <property name="styleSheet">
           <string notr="true">QTextBrowser {
    background: rgba(0, 0, 0, 180);
    border: 2px solid #555;
    border-radius: 8px;
    padding: 15px;
    color: #E0E0E0;
    font-size: 16px;
    font-family: 'Smiley Sans',oblique;
}
QScrollBar:vertical {
    background: rgba(0, 0, 0, 100);
    width: 12px;
    border-radius: 6px;
}
QScrollBar::handle:vertical {
    background: #00BCD4;
    border-radius: 6px;
    min-height: 20px;
}</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonLayout">
     <property name="spacing">
      <number>50</number>
     </property>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="HoverButton" name="returnButton">
       <property name="minimumSize">
        <size>
         <width>180</width>
         <height>60</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
    stop:0 #FF5722, stop:1 #FF9800);
    color: white;
    border: none;
    border-radius: 12px;
    font: bold 18px 'Smiley Sans';
    padding: 15px 30px;
}
QPushButton:hover {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
    stop:0 #FF8A65, stop:1 #FFB74D);
    border: 2px solid white;
}
QPushButton:pressed {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
    stop:0 #E64A19, stop:1 #F57C00);
}</string>
       </property>
       <property name="text">
        <string>Return</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="HoverButton" name="queryButton">
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>60</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
    stop:0 #2196F3, stop:1 #00BCD4);
    color: white;
    border: none;
    border-radius: 12px;
    font: bold 18px 'Smiley Sans';
    padding: 15px 30px;
}
QPushButton:hover {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
    stop:0 #64B5F6, stop:1 #4DD0E1);
    border: 2px solid white;
}
QPushButton:pressed {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
    stop:0 #1976D2, stop:1 #0097A7);
}</string>
       </property>
       <property name="text">
        <string>Query Rankings</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HoverButton</class>
   <extends>QPushButton</extends>
   <header>hoverbutton.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
